micrometer-spring-legacy/src/samples/java/io/micrometer/spring/samples/library/ProposalLibrary.java

```diff
@@ -0,0 +1,18 @@
+package io.micrometer.spring.samples.library;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Meters;
+
+import javax.inject.Inject;
+
+public class ProposalLibrary {
+
+    @Inject MeterRegistry registry = MeterRegistry.globalRegistry;
```

Comment:
There is a potential con here that users of my library need to know which of my components to bind with their DI framework.  Or, I have to provide a guice module or equivalent for each DI framework I intend to support.